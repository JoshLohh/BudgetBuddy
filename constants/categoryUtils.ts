import { Ionicons } from "@expo/vector-icons";

export const CATEGORIES = [
  { label: 'Food', value: 'Food', icon: 'restaurant' },
  { label: 'Transportation', value: 'Transportation', icon: 'car' },
  { label: 'Accommodation', value: 'Accommodation', icon: 'home' },
  { label: 'Entertainment', value: 'Entertainment', icon: 'film' },
  { label: 'Utilities', value: 'Utilities', icon: 'bulb' },
  { label: 'Shopping', value: 'Shopping', icon: 'cart' },
  { label: 'Others', value: 'Others', icon: 'ellipsis-horizontal' },
];

export const getCategoryIconName = (category: string) => {
  const found = CATEGORIES.find(c => c.value === category);
  return found?.icon || 'ellipsis-horizontal';
};

export type IoniconName = React.ComponentProps<typeof Ionicons>['name'];