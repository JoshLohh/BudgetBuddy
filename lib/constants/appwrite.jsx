import { Client, Account, ID, Avatars } from 'react-native-appwrite';


export const client = new Client
    .setProject("6838ad0d003d2a0754a0")
    .setPlatform('dev.Buddies.budgetbuddy');

export const account = new Account(client)
export const avatars = new Avatars(client)