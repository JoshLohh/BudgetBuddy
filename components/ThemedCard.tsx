import { useThemeColor } from '@/hooks/useThemeColor';
import { StyleSheet, View , ViewProps, useColorScheme } from 'react-native';
import { Colors } from '../constants/Colors'

const ThemedCard = ({ style, ...props }: ViewProps) => {
    const colorScheme = useColorScheme()
    const theme = Colors[colorScheme === "dark" ? "dark" : "light" ] ?? Colors.light
  
    return (
        <View
            style = {[
                {
                    backgroundColor: theme.uiBackGround,
                    padding: 20,
                    borderRadius: 5,
                },
                style
            ]}
            {...props}
        />
  )

}

export default ThemedCard