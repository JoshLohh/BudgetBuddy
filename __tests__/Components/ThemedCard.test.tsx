import React from 'react';
import { render } from '@testing-library/react-native';
import * as ReactNative from 'react-native';
import ThemedCard from '../../components/ThemedCard';
import { Colors } from '../../constants/Colors';

describe('ThemedCard', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('renders correctly with light theme background', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('light');
    const { getByTestId } = render(<ThemedCard testID="card" />);
    const card = getByTestId('card');
    const styles = Array.isArray(card.props.style) ? card.props.style : [card.props.style];

    expect(styles).toEqual(
      expect.arrayContaining([
        expect.objectContaining({ backgroundColor: Colors.light.uiBackGround }),
      ]),
    );
  });

  it('renders correctly with dark theme background', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('dark');
    const { getByTestId } = render(<ThemedCard testID="card" />);
    const card = getByTestId('card');
    const styles = Array.isArray(card.props.style) ? card.props.style : [card.props.style];

    expect(styles).toEqual(
      expect.arrayContaining([
        expect.objectContaining({ backgroundColor: Colors.dark.uiBackGround }),
      ]),
    );
  });

  it('defaults to light theme when scheme is undefined', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue(undefined);
    const { getByTestId } = render(<ThemedCard testID="card" />);
    const card = getByTestId('card');
    const styles = Array.isArray(card.props.style) ? card.props.style : [card.props.style];

    expect(styles).toEqual(
      expect.arrayContaining([
        expect.objectContaining({ backgroundColor: Colors.light.uiBackGround }),
      ]),
    );
  });

  it('applies additional custom styles', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('light');
    const extraStyle = { margin: 20, padding: 99 };
    const { getByTestId } = render(<ThemedCard testID="card" style={extraStyle} />);
    const card = getByTestId('card');
    const styles = Array.isArray(card.props.style) ? card.props.style : [card.props.style];

    expect(styles).toEqual(
      expect.arrayContaining([
        expect.objectContaining(extraStyle),
      ]),
    );
  });

  it('matches the snapshot for light theme', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('light');
    const tree = render(<ThemedCard testID="card" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches the snapshot for dark theme', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('dark');
    const tree = render(<ThemedCard testID="card" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

/**
 * This test file verifies that ThemedCard renders with the correct UI background
 * color (from Colors.light.uiBackGround or Colors.dark.uiBackGround), depending on the
 * color scheme. It also checks proper merging of custom styles, and uses snapshot
 * tests to monitor unintended output changes for both themes.
 */
