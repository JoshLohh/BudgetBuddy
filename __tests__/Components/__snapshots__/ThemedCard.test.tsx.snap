// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedCard matches the snapshot for dark theme 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#2f2b3d",
        "borderRadius": 5,
        "padding": 20,
      },
      undefined,
    ]
  }
  testID="card"
/>
`;

exports[`ThemedCard matches the snapshot for light theme 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#d6d5e1",
        "borderRadius": 5,
        "padding": 20,
      },
      undefined,
    ]
  }
  testID="card"
/>
`;
